api/envoy/api/v2/core/base.proto

```diff
@@ -259,7 +259,7 @@ message SocketOption {
   // The state in which the option will be applied. When used in BindConfig
   // STATE_PREBIND is currently the only valid value.
   SocketState state = 6
-      [(validate.rules).message.required = true, (validate.rules).enum.defined_only = true];
+      [(validate.rules).any.required = true, (validate.rules).enum.defined_only = true];
```

Comment:
hmm, remove `(validate.rules).any.required = true`? SocketState is an enum and that should do nothing.

Reply:
I think that’s still required to ensure the field is not, not provided.

Any required ensure a value is provided. And then enum defined_only ensures it’s not bogus

Reply:
No, proto3 doesn't, and have no way to, differentiate non-set vs default-value for non-message types so it is no-op. https://developers.google.com/protocol-buffers/docs/proto3#default

Reply:
Ah you're right my bad. I'll push up another commit that removes it. Might be worth opening up a ticket to find other bad uses of required.